src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsHtmlReport.java

```diff
@@ -53,33 +63,45 @@ public String getName() {
 
   @Override
   public TextResource getStylesheet() {
-    return stylesheet.getOrNull();
-  }
+    if (!stylesheet.isPresent() && !stylesheetPath.isPresent()) {
+      return super.getStylesheet();
+    }
 
-  @Override
-  public void setStylesheet(@Nullable TextResource textResource) {
-    stylesheet.set(textResource);
-  }
+    if (stylesheet.isPresent()) {
+      return stylesheet.get();
+    }
 
-  @Override
-  public void setStylesheet(@Nullable String path) {
-    Optional<File> spotbugsJar =
+    TextResourceFactory factory = handler.getText();
+    Optional<File> spotbugs =
         getTask().getProject().getConfigurations().getByName("spotbugs")
-            // FIXME this operation probably evaluates the spotbugs configuration, that may make the
-            // build slow
             .files(
                 dependency ->
                     dependency.getGroup().equals("com.github.spotbugs")
                         && dependency.getName().equals("spotbugs"))
             .stream()
             .findFirst();
-    if (spotbugsJar.isPresent()) {
-      TextResource textResource =
-          getTask().getProject().getResources().getText().fromArchiveEntry(spotbugsJar.get(), path);
-      setStylesheet(textResource);
+    if (spotbugs.isPresent()) {
+      File jar = spotbugs.get();
+      logger.debug(
+          "Specified stylesheet ({}) found in spotbugs configuration: {}",
```

Comment:
At this moment we're not sure that the spotbugs jar file contains specified stylesheet or not, right?
I feel this log message is not so intuitive.